qutip/qip/circuit_latex.py

```diff
@@ -42,31 +47,121 @@
 """
 
 
-def _latex_compile(code, filename="qcirc", format="png"):
+def _find_system_command(names):
+    for name in names:
+        if shutil.which(name) is not None:
+            return name
+    return None
+
+
+_pdflatex = _find_system_command(['pdflatex'])
+_pdfcrop = _find_system_command(['pdfcrop'])
+
+
+def _force_remove(*filenames):
+    """rm -f`: try to remove a file, ignoring errors if it doesn't exist."""
+    for filename in filenames:
+        try:
+            os.remove(filename)
+        except FileNotFoundError:
+            pass
+
+
+_run_command = functools.partial(subprocess.run, check=True,
+                                 stdout=subprocess.DEVNULL,
+                                 stderr=subprocess.DEVNULL)
+_run_command.__doc__ = \
+    """
+    Run a command with stdout and stderr explicitly thrown away, raising
+    `subprocess.CalledProcessError` if the command returned a non-zero exit
+    code.
+    """
+
+if not _pdfcrop:
```

Comment:
Better use `if _pdfcrop is None:`?

Reply:
I would say `if not x` is pythonic and understandable - the alternative is a string with the system command name in it, and the python coding guidelines suggest using `if not x` to check for empty strings.

Reply:
Looking again, it's pretty questionable why I'm inconsistent within the same block of code - I do `if which is None` later on which is exactly the same case.